drivers/include/mfrc522.h

```diff
@@ -0,0 +1,844 @@
+
+/**
+ * @brief Performs the "magic sequence" needed to get Chinese UID changeable
+ *        Mifare cards to allow writing to sector 0, where the card UID is stored.
+ *
+ * @note  Note that you do not need to have selected the card through REQA or WUPA,
+ *        this sequence works immediately when the card is in the reader vicinity.
+ *        This means you can use this method even on "bricked" cards that your
+ *        reader does not recognise anymore (see mfrc522_mifare_unbrick_uid_sector).
+ *        Of course with non-bricked devices, you are free to select them before
+ *        calling this function.
+ *
+ * @param[in] dev  Device descriptor of the MFRC522
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_mifare_open_uid_backdoor(mfrc522_t *dev);
+
+/**
+ * @brief Read entire block 0, including all manufacturer data, and overwrites
+ *        that block with the new UID, a freshly calculated BCC, and the original
+ *        manufacturer data.
+ *
+ * @note  It assumes a default KEY A of 0xFFFFFFFFFFFF. Make
+ *        sure to have selected the PICC before this function is called and that
+ *        the UID of the PICC is changeable.
+ *
+ * @param[in] dev           Device descriptor of the MFRC522
+ * @param[in] uid           PICC's UID
+ * @param[in] new_uid       New UID to set on PICC
+ * @param[in] new_uid_size  Size of new UID
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_mifare_set_uid(mfrc522_t *dev,
+                                             mfrc522_uid_t *uid,
+                                             uint8_t *new_uid,
+                                             uint8_t new_uid_size);
+
+/**
+ * @brief Reset entire sector 0 to zeroes, so the card can be read again by readers
+ *
+ * @param[in] dev  Device descriptor of the MFRC522
+ *
+ * @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_mifare_unbrick_uid_sector(mfrc522_t *dev);
+
+/**
+ * @brief Checks whether a new card could be detected
+ *
+ * @note  Only "new" cards in state IDLE are invited. Sleeping cards in state
+ *        HALT are ignored.
+ *
+ * @param[in] dev  Device descriptor of the MFRC522
+ *
+ * @return true if new card was detected, false otherwise
+ */
+bool mfrc522_picc_is_new_card_present(mfrc522_t *dev);
+
+/**
+ * @brief Simple wrapper around mfrc522_picc_select()
+ *
+ * @note  Remember to call mfrc522_picc_is_new_card_present(),
+ *        mfrc522_picc_request_a() or mfrc522_picc_wakeup_a() first.
+ *
+ * @param[in] dev  Device descriptor of the MFRC522
+ * @param[in] uid  PICC's UID
+ *
+ * @return @return MFRC522_STATUS_OK on success, MFRC522_STATUS_??? otherwise
+ */
+mfrc522_status_code_t mfrc522_picc_read_card_serial(mfrc522_t *dev,
+                                                    mfrc522_uid_t *uid);
+
+#ifdef MFRC522_DUMP_FUNCTIONS
```

Comment:
Is this actually needed? If functions are not used I assume they would not be linked. What's the advantage?

Reply:
Also, in case we keep it for some reason:
- Make it a pseudomodule instead of a `CFLAG`
- OR the macro with `DOXYGEN` so it is included in the documentation

Reply:
It once had an advantage, but not anymore. I removed the `ifdef`s